File: src/main/java/de/steklopod/sec08/Lec03Merge.java
package de.steklopod.sec08;

import de.steklopod.utils.Util;
import de.steklopod.sec08.helper.AmericanAirlines;
import de.steklopod.sec08.helper.Emirates;
import de.steklopod.sec08.helper.Qatar;
import reactor.core.publisher.Flux;

public class Lec03Merge {

    public static void main(String[] args) {

        Flux<String> merge = Flux.merge(
                Qatar.getFlights(),
                Emirates.getFlights(),
                AmericanAirlines.getFlights()
        );

        merge.subscribe(Util.subscriber());

        Util.sleepSeconds(10);

    }


}

File: src/main/kotlin/de/steklopod/sec08/Lec03Merge.kt
package de.steklopod.sec08

import de.steklopod.sec08.helper.AmericanAirlines
import de.steklopod.sec08.helper.Emirates
import de.steklopod.sec08.helper.Qatar
import de.steklopod.utils.Util.sleepSeconds
import de.steklopod.utils.Util.subscriber
import reactor.core.publisher.Flux

object Lec03Merge {
    @JvmStatic
    fun main(args: Array<String>) {
        val merge = Flux.merge(
            Qatar.flights,
            Emirates.flights,
            AmericanAirlines.flights
        )
        merge.subscribe(subscriber())
        sleepSeconds(10)
    }
}
